kevm-pyk/src/kevm_pyk/solc_to_k.py

```diff
@@ -543,38 +543,78 @@ def _evm_base_sort_int(type_label: str) -> bool:
 
 
 def _range_predicate(term: KInner, type_label: str) -> KInner | None:
-    (success, result) = _range_predicate_uint(term, type_label)
-    if success:
-        return result
-    if type_label == 'address':
-        return KEVM.range_address(term)
-    if type_label == 'bool':
-        return KEVM.range_bool(term)
-    if type_label == 'bytes4':
-        return KEVM.range_bytes(intToken(4), term)
-    if type_label in {'bytes32', 'uint256'}:
-        return KEVM.range_uint(256, term)
-    if type_label == 'int256':
-        return KEVM.range_sint(256, term)
-    if type_label == 'bytes':
-        return KEVM.range_uint(128, KEVM.size_bytes(term))
-    if type_label == 'string':
-        return TRUE
+    match type_label:
+        case 'address':
+            return KEVM.range_address(term)
+        case 'bool':
+            return KEVM.range_bool(term)
+        case 'bytes':
+            return KEVM.range_uint(128, KEVM.size_bytes(term))
+        case 'string':
+            return TRUE
+
+    predicate_functions = [
+        _range_predicate_uint,
+        _range_predicate_int,
+        _range_predicate_bytes,
+    ]
+
+    for f in predicate_functions:
+        (success, result) = f(term, type_label)
+        if success:
+            return result
 
     _LOGGER.info(f'Unknown range predicate for type: {type_label}')
     return None
 
 
 def _range_predicate_uint(term: KInner, type_label: str) -> tuple[bool, KInner | None]:
     if type_label.startswith('uint') and not type_label.endswith(']'):
-        width = int(type_label[4:])
+        if type_label == 'uint':
+            width = 256
```

Comment:
So are you automatically set a witdh of `256` for all `uints`? What happens to `uint8` for example?

Reply:
Not really, it should only set the width of `256` if the type label is strictly `uint`, because it's a shorthand of `uint256`. If it's `uint256` or `uint8` it should fall in the `else` clause where the `width` is dynamically set.